libraries/psio/src/schema.cpp

```diff
@@ -152,12 +154,84 @@ namespace psio::schema_types
       }
    };
 
+   template <int Digits>
+   struct TimePointImpl
+   {
+      // Unwrap a single element struct
+      static const CompiledType* unwrap(const CompiledType* type)
+      {
+         if (type->kind == CompiledType::struct_ && type->children.size() == 1 &&
+             !type->children[0].is_optional)
+            return type->children[0].type;
+         else
+            return type;
+      }
+      static bool match(const CompiledType* type)
+      {
+         type = unwrap(type);
+         // Accept integers up to 64 bits
+         if (type->kind == CompiledType::scalar)
+         {
+            if (auto* itype = std::get_if<Int>(&type->original_type->value))
+            {
+               return itype->bits <= 64;
+            }
+         }
+         return false;
+      }
+      static bool frac2json(const CompiledType* type, FracStream& in, StreamBase& out)
+      {
+         type = unwrap(type);
+         if (auto* itype = std::get_if<Int>(&type->original_type->value))
+         {
+            // read up to 8 bytes
+            std::uint64_t raw = 0;
+            for (int i = 0; i < (itype->bits + 7) / 8; ++i)
+            {
+               std::uint8_t digit;
+               if (!in.unpack<true, true>(&digit))
+                  return false;
+               raw += digit << (i * 8);
```

Comment:
Shouldn't `digit` be casted to `uint64_t` before shifting?

Reply:
Yes, it promotes to int, but that's not enough.